calibration_assistant.cpp

```diff
@@ -0,0 +1,1247 @@
+        {
+            double expectedRatio = cos(newCal.declination);
+            if (newDetails.raGuideSpeed > 0.)                   // for mounts that may have different guide speeds on RA and Dec axes
+                speedRatio = newDetails.decGuideSpeed / newDetails.raGuideSpeed;
+            else
+                speedRatio = 1.0;
+            actualRatio = newCal.xRate * speedRatio / newCal.yRate;
+            ratesMeaningful = true;
+            debugVals += wxString::Format("Spds: %.1fX,%.1fX, ", RateX(newDetails.raGuideSpeed), RateX(newDetails.decGuideSpeed)) +
+                wxString::Format("Dec: %.1f, Rates: %.1f, %.1f, ", degrees(newCal.declination), RateX(newCal.xRate), RateX(newCal.yRate));
+        }
+        else
+            debugVals += "Spds: N/A, ";
+
+        goodRslt = (newDetails.raStepCount >= 2 * CAL_ALERT_MINSTEPS || (newDetails.decStepCount >= 2 * CAL_ALERT_MINSTEPS && newDetails.decStepCount > 0)) &&         // Dec guiding might be disabled
+            (newDetails.raStepCount <= 30 || (newDetails.decStepCount <= 30 && newDetails.decStepCount > 0));       // Not too few, not too many
+        debugVals += wxString::Format("Steps: %d,%d, ", newDetails.raStepCount, newDetails.decStepCount);
+        if (!goodRslt)
+            evalWhy = _("Steps, ");
+
+        // Non-orthogonal RA/Dec axes. Values in Calibration structures are in radians
+        double nonOrtho = degrees(fabs(fabs(norm_angle(newCal.xAngle - newCal.yAngle)) - M_PI / 2.));         // Delta from the nearest multiple of 90 degrees
+        debugVals += wxString::Format("Ortho: %.2f, ", nonOrtho);
+        if (nonOrtho > 5)
+        {
+            evalWhy += _("Orthogonality, ");
+            goodRslt = false;
+            acceptableRslt = acceptableRslt && (nonOrtho <= CAL_ALERT_ORTHOGONALITY_TOLERANCE);
+        }
+        if (ratesMeaningful)
+        {
+            debugVals += wxString::Format("Rates: %.2f (Expect) vs %.2f (Act)", cos(newCal.declination), actualRatio);
+            if (fabs(cos(newCal.declination) - actualRatio) > 0.1)
+            {
+                evalWhy += _("Rates, ");
+                goodRslt = false;
+                acceptableRslt = acceptableRslt && (expectedRatio - actualRatio) < CAL_ALERT_AXISRATES_TOLERANCE;
+            }
+        }
+        if (fabs(degrees(newCal.declination)) > 60)
+        {
+            goodRslt = false;
+            acceptableRslt = false;
+            evalWhy += _("Sky location (Dec comp disabled), ");;
+        }
+        else if (fabs(degrees(newCal.declination)) > 20)
+        {
+            goodRslt = false;
+            evalWhy += _("Sky location, ");
+        }
+
+        evalWhy = "(" + evalWhy.Truncate(evalWhy.length() - 2) + ")";
+        m_lastResult = evalWhy;
+        Debug.Write(debugVals + "\n");
+
+        if (goodRslt)
+        {
+            ShowStatus(_("Calibration produced a good result!"));

```

Comment:
All I wanted here was a 'thumbs-up' message, didn't want to get into TMI.  People don't like calibration.  Setting auto-restore really depends on whether they set up every night and are being careful to reassemble the guiding assembly the same way each night.  Even the "any sky location" behavior depends on having Dec comp set - admittedly nearly always true.  I can be convinced if you feel strongly about it but sometimes less said is more.

Reply:
agreed, don't want to fall int a TMI trap, which my suggested message probably does.
I am imagining the newbie person running the calibration wiz and getting this message, and then wondering "now what?".   Was trying to suggest we tell them what to do next. Any better ideas?

Reply:
At this point, guiding has started.  I suppose I could say guiding has started using the new calibration.  I didn't want to close the dialog because the user might want to iterate or experiment further and I wanted them to see a ta-da message.